src/main/java/com/jnape/palatable/shoki/api/MultiSet.java

```diff
@@ -107,4 +112,137 @@
     default MultiSet<A> addAll(MultiSet<A> other) {
         return foldLeft(curried(ms -> into(ms::add)), this, other);
     }
+
+
+    /**
+     * {@inheritDoc}
+     * Amortized <code>O(1)</code>.
+     */
+    @Override
+    default boolean contains(A a) {
+        return contains(a, one());
+    }
+
+    /**
+     * Determine if some value <code>k</code> <code>a</code>s are elements of this {@link MultiSet}.
+     * Equivalent to <code>get(a) >= k</code>
+     *
+     * @param a the value
+     * @param k the amount of a
+     * @return true if there are k <code>a</code>s in this {@link MultiSet}; false otherwise
+     */
+    default boolean contains(A a, NonZero k) {
+        return gte(k, get(a));
+    }
+
+    /**
+     * The union of two {@link MultiSet MultiSets}
+     * <code>xs</code> and <code>ys</code> is the {@link MultiSet} of those elements present in either <code>xs</code>
+     * or <code>ys</code> with the occurrences of whichever one had more elements. This is equivalent to
+     * {@link MultiSet#merge} partially applied with {@link com.jnape.palatable.lambda.semigroup.builtin.Max#max}.
+     *
+     * @param other the {@link MultiSet} to union with this {@link MultiSet}
+     * @return the union {@link MultiSet}
+     * @see MultiSet#merge
+     */
+    default MultiSet<A> union(MultiSet<A> other) {
+        return merge(max(), other);
+    }
+
+    /**
+     * The <a href="https://en.wikipedia.org/wiki/Complement_(set_theory)#Relative_complement"
+     * target="_new">difference</a> (also known as "relative complement") of two {@link MultiSet MultiSets}
+     * <code>xs</code> and <code>ys</code> is the {@link MultiSet} of in <code>xs</code> with all elements of
+     * <code>ys</code> removed.
+     *
+     * @param other the {@link MultiSet} to subtract from this {@link MultiSet}
+     * @return the difference {@link MultiSet}
+     */
+    default MultiSet<A> difference(MultiSet<A> other) {
+        return foldLeft(curried(ms -> into(ms::remove)), this, other);
+    }
+
+    /**
+     * Determine if another {@link MultiSet} is a subset of this {@link MultiSet}. A {@link MultiSet} <code>x</code> is
+     * a subset of <code>y</code> if for every <code>a^i</code> in <code>x</code> there exists <code>a^j</code> in
+     * <code>y</code> where ^ denotes the number of occurrences in the multiset.
```

Comment:
Futile attempt to use a notation for occurrences.

Reply:
The notation is a challenge. I went with `(a * k)` in the `toString`. It's annoying. I don't think I like the `xor`/exponentiation notation but it won't prevent a merge, so don't sweat it.

Reply:
I shamelessly stole it from wikipedia and a couple papers.